src/targets/maven.ts

```diff
@@ -444,4 +391,105 @@ export class MavenTarget extends BaseTarget {
 
     return `${moduleName}.jar`;
   }
+
+  // Maven central does not indicate when it completes the action, so we need to
+  // retry every so often and query it for the new state of repository.
+  // Based on: https://github.com/vanniktech/gradle-maven-publish-plugin/ implementation.
+  public async closeAndReleaseRepository(): Promise<void> {
+    const repository = await this.getRepository();
+    const { repositoryId, type } = repository;
+
+    if (type !== 'open') {
+      throw new Error(
+        'No open repositories available. Go to Nexus Repository Manager to see what happened.'
+      );
+    }
+
+    await this.closeRepository(repositoryId);
+    await this.releaseRepository(repositoryId);
+  }
+
+  private getNexusRequestHeaders(): Record<string, string> {
+    return {
+      Accept: 'application/json',
+      Authorization: `Basic ${Buffer.from(
+        `${process.env.OSSRH_USERNAME}:${process.env.OSSRH_USERNAME}`
+      ).toString(`base64`)}`,
+    };
+  }
+
+  private async getRepository(): Promise<NexusRepository> {
+    const response = await fetch(`${NEXUS_API_BASE_URL}/profile_repositories`, {
+      headers: this.getNexusRequestHeaders(),
+    });
+    if (!response.ok) {
+      throw new Error(
+        `Unable to fetch repository: ${response.status}, ${response.statusText}`
+      );
+    }
+
+    const body = await response.json();
+    const repositories = body.data;
+
+    if (repositories.length === 0) {
+      throw new Error(`No available repositories. Nothing to publish.`);
```

Comment:
nit:
```suggestion
      throw new Error(`No available repositories. Nothing to publish to.`);
```
?

Reply:
We publish "a thing" (artifacts already present at this repository), not "to something", so I believe this is correct?

Reply:
Yes, we publish artifacts; but, we throw the error when there's no repository to publish the artifacts to. The error is not about the artifacts, but about where we publish them. This is clear in the first sentence, and thus a bike-shedding discussion, so your call.